sycl/source/detail/program_impl.cpp

```diff
@@ -291,9 +291,13 @@ void program_impl::link(string_class LinkOptions) {
     check_device_feature_support<info::device::is_linker_available>(MDevices);
     vector_class<RT::PiDevice> Devices(get_pi_devices());
     const detail::plugin &Plugin = getPlugin();
+    const char *LinkOpts = SYCLConfig<SYCL_PROGRAM_LINK_OPTIONS>::get();
+    if (!LinkOpts) {
+      LinkOpts = LinkOptions.c_str();
+    }
     RT::PiResult Err = Plugin.call_nocheck<PiApiKind::piProgramLink>(
-        MContext->getHandleRef(), Devices.size(), Devices.data(),
-        LinkOptions.c_str(), 1, &MProgram, nullptr, nullptr, &MProgram);
+        MContext->getHandleRef(), Devices.size(), Devices.data(), LinkOpts, 1,
```

Comment:
Would be great to add a comment, explaining the meaning of `1`, even though it is not present now.

Reply:
Why only this parameter? There are nowhere in the file another comments about parameters in the pi functions

Reply:
1 is some kind of magic value here. Looking at the diff I don’t really understand what it means. And while I agree there are no other places in the file, where such constants are commented somehow, I think it’s a good idea to put something like this, since you’re changing the code anyway.
```suggestion
        MContext->getHandleRef(), Devices.size(), Devices.data(), LinkOpts, /* num_input_programs*/ 1,
```

Reply:
Well, I'll add that comment